aidial_adapter_bedrock/llm/consumer.py

```diff
@@ -42,23 +42,36 @@ def add_usage(self, usage: TokenUsage):
     def set_discarded_messages(self, discarded_messages: List[int]):
         pass
 
+    @abstractmethod
+    def create_function_tool_call(self, tool_call: ToolCall):
+        pass
+
 
 class ChoiceConsumer(Consumer):
     usage: TokenUsage
     choice: Choice
     discarded_messages: Optional[List[int]]
-    tools_emulator: ToolsEmulator
+    tools_emulator: Optional[ToolsEmulator]
 
-    def __init__(self, tools_emulator: ToolsEmulator, choice: Choice):
+    def __init__(
+        self,
+        choice: Choice,
+    ):
         self.choice = choice
         self.usage = TokenUsage()
         self.discarded_messages = None
+        self.tools_emulator = None
+
+    def add_emulator(self, tools_emulator: ToolsEmulator):
```

Comment:
Maybe `set_tools_emulator` is better, because it's basically a setter _(and also in line with existing `set_discarded_messages` method)_.

`add_X` typically implies that calling it multiple times isn't the same as calling it once.